### ISSUE TITLE
[TW-1745] Allow for use of environment variables in config file

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1769]
_Alan Young on 2015-12-30T17:20:56Z says:_ 

 The specific example I want to use is, in the .taskrc file:

include data.location/include/file_to.include

where data.location is replaced with the value of the config option data.location.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-12-30T17:20:56Z
Modified: 2017-01-16T18:29:11Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2017-01-16T18:29:11Z says:_ 

 Change title to add 'environment' which might be more useful.

**Comment by dsifford** [comment]
Just adding some comments related to what the last guy (directly above me) said:

Simply "using" or reading environment variables in `.taskrc` is also something I'd be interested in...

Specific use-case:

I have `TASKRC` in my environment pointed to `"$XDG_CONFIG_HOME"/task/taskrc` and inside my taskrc I would like to set `data.location` to `"$XDG_CONFIG_HOME/task"`.

**Comment by Neurognostic** [comment]
Another use case: I would like to be able to read environment variables for configuring certificate, key, and ca paths for `taskd` syncing.

Example, as of version `2.5.1`, this does not currently work:

```config
taskd.certificate=$XDG_DATA_HOME/task/taskd.cert.pem
taskd.key=$XDG_DATA_HOME/task/taskd.key.pem
taskd.ca=$XDG_DATA_HOME/task/taskd.ca.pem
```

But taskwarrior does understand `~` means `$HOME` so this does work:

```config
taskd.certificate=~/.local/share/task/taskd.cert.pem
taskd.key=$XDG_DATA_HOME/task/taskd.key.pem
taskd.ca=$XDG_DATA_HOME/task/taskd.ca.pem
```

**Comment by tbabej** [comment]
Implemented via #2558.
